corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -829,53 +830,49 @@ def get_case_ids_in_domain_by_owners(domain, owner_ids, closed=None):
     def save_case(case):
-        transactions_to_save = case.get_tracked_models_to_create(CaseTransaction)
+        db_name = get_db_alias_for_partitioned_doc(case.case_id)
+        transactions_to_save = case.get_live_tracked_models(CaseTransaction)
 
         indices_to_save_or_update = case.get_live_tracked_models(CommCareCaseIndexSQL)
         index_ids_to_delete = [index.id for index in case.get_tracked_models_to_delete(CommCareCaseIndexSQL)]
 
         attachments_to_save = case.get_tracked_models_to_create(CaseAttachmentSQL)
         attachment_ids_to_delete = [att.id for att in case.get_tracked_models_to_delete(CaseAttachmentSQL)]
 
-        for index in indices_to_save_or_update:
-            index.domain = case.domain  # ensure domain is set on indices
-
-        # cast arrays that can be empty to appropriate type
-        query = """SELECT case_pk FROM save_case_and_related_models(
-            %s, %s, %s, %s::{}[], %s::{}[], %s::INTEGER[], %s::INTEGER[]
-        )"""
-        query = query.format(CommCareCaseIndexSQL_DB_TABLE, CaseAttachmentSQL_DB_TABLE)
-        with get_cursor(CommCareCaseSQL) as cursor:
-            try:
-                cursor.execute(query, [
-                    case.case_id,
-                    case,
-                    transactions_to_save,
-                    indices_to_save_or_update,
-                    attachments_to_save,
-                    index_ids_to_delete,
-                    attachment_ids_to_delete
-                ])
-                result = fetchone_as_namedtuple(cursor)
-                case.id = result.case_pk
-            except InternalError as e:
-                if logging.root.isEnabledFor(logging.DEBUG):
-                    msg = 'save_case_and_related_models called with args: \n{}, {}, {}, {} ,{} ,{}'.format(
-                        case_adapter(case).getquoted(),
-                        [case_transaction_adapter(t).getquoted() for t in transactions_to_save],
-                        [case_index_adapter(i).getquoted() for i in indices_to_save_or_update],
-                        [case_attachment_adapter(a).getquoted() for a in attachments_to_save],
-                        index_ids_to_delete,
-                        attachment_ids_to_delete
-                    )
-                    logging.debug(msg)
-                raise CaseSaveError(e)
-            else:
+        try:
+            with transaction.atomic(using=db_name):
+                case.save(using=db_name)
+                for case_transaction in transactions_to_save:
+                    case_transaction.save(using=db_name)
+
+                for index in indices_to_save_or_update:
+                    index.domain = case.domain  # ensure domain is set on indices
+                    update_fields = None
+                    if index.is_saved():
+                        # prevent changing identifier
+                        update_fields = ['referenced_id', 'referenced_type', 'relationship_id']
```

Comment:
good question, not 100% sure but it's functionality we support now so don't want to remove it just yet.